driver/src/test/java/org/neo4j/driver/ConfigTest.java

```diff
@@ -348,4 +361,73 @@ void shouldErrorWithInvalidUserAgent()
         assertThrows( IllegalArgumentException.class, () -> Config.builder().withUserAgent( null ).build() );
         assertThrows( IllegalArgumentException.class, () -> Config.builder().withUserAgent( "" ).build() );
     }
+
+    @Nested
+    class SerializationTest
+    {
+
+        @Test
+        void shouldSerialize() throws Exception
+        {
+            Config config = Config.builder().withMaxConnectionPoolSize( 123 ).withConnectionTimeout( 6543L, TimeUnit.MILLISECONDS ).withConnectionAcquisitionTimeout(
+                    5432L, TimeUnit.MILLISECONDS ).withConnectionLivenessCheckTimeout( 4321L, TimeUnit.MILLISECONDS ).withMaxTransactionRetryTime( 3210L, TimeUnit.MILLISECONDS ).withFetchSize( 9876L ).withEventLoopThreads(
+                    4 ).withoutEncryption().withTrustStrategy( Config.TrustStrategy.trustAllCertificates() ).withUserAgent( "user-agent" ).withDriverMetrics().withRoutingTablePurgeDelay( 50000, TimeUnit.MILLISECONDS ).withLeakedSessionsLogging().build();
+
+            Config verify = TestUtil.serializeAndReadBack( config, Config.class );
+
+
+            assertEquals( config.maxConnectionPoolSize(), verify.maxConnectionPoolSize() );
+            assertEquals( config.connectionTimeoutMillis(), verify.connectionTimeoutMillis() );
+            assertEquals( config.connectionAcquisitionTimeoutMillis(), verify.connectionAcquisitionTimeoutMillis() );
+            assertEquals( config.idleTimeBeforeConnectionTest(), verify.idleTimeBeforeConnectionTest() );
+            assertEquals( config.maxConnectionLifetimeMillis(), verify.maxConnectionLifetimeMillis() );
+            assertNotNull( verify.retrySettings() );
+            assertSame( DevNullLogging.DEV_NULL_LOGGING, verify.logging() );
+            assertEquals( config.retrySettings().maxRetryTimeMs(), verify.retrySettings().maxRetryTimeMs() );
+            assertEquals( config.fetchSize(), verify.fetchSize() );
+            assertEquals( config.eventLoopThreads(), verify.eventLoopThreads() );
+            assertEquals( config.encrypted(), verify.encrypted() );
+            assertEquals( config.trustStrategy().strategy(), verify.trustStrategy().strategy() );
+            assertEquals( config.trustStrategy().isHostnameVerificationEnabled(), verify.trustStrategy().isHostnameVerificationEnabled() );
+            assertEquals( config.trustStrategy().revocationStrategy(), verify.trustStrategy().revocationStrategy() );
```

Comment:
Might be worth covering `certFile()` too.

Reply:
File is serializable.